File: nowcoder/C++/BM1.反转链表.cc
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
        ListNode* pre = nullptr;
        ListNode* cur = pHead;
        while (cur != nullptr) {
            ListNode* lat = cur->next;
            cur->next = pre;
            pre = cur;
            cur = lat;
        }
        return pre;
    }
};

File: nowcoder/Golang/BM1.反转链表.go
package nowcoder

// import . "nc_tools"
type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 *
 * @param pHead ListNode类
 * @return ListNode类
 */
func ReverseList(pHead *ListNode) *ListNode {
	var pre *ListNode = nil
	var cur *ListNode = pHead
	for cur != nil {
		var lat = cur.Next
		cur.Next = pre
		pre = cur
		cur = lat
	}
	return pre
}
